lib/IRGen/GenClangDecl.cpp

```diff
@@ -120,6 +216,29 @@ void IRGenModule::emitClangDecl(const clang::Decl *decl) {
     stack.push_back(D);
   });
 
+  std::vector<const clang::Decl *> ctors;
+  llvm::copy_if(stack, std::back_inserter(ctors), [](const clang::Decl *decl) {
+    return isa<clang::CXXConstructorDecl>(decl);
+  });
+
+  std::set<const clang::Decl *> visited;
+  std::vector<const clang::Decl *> functionDecls;
+  for (auto *ctor : ctors)
+    walkCallGraphFromCtor(cast<clang::CXXConstructorDecl>(ctor), visited,
+                          functionDecls);
+  llvm::sort(functionDecls);
+  functionDecls.erase(std::unique(functionDecls.begin(), functionDecls.end()),
```

Comment:
We really want this to factor in the existing stack when sorting out duplicate decls, right?

Reply:
I wasn't sure of that. I did not want to mess with the existing top level decls that are originally given to emitClangDecl. I could.